vcd/resource_vcd_rights_bundle.go

```diff
@@ -0,0 +1,313 @@
+			},
+			"read_only": &schema.Schema{
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Whether this rights bundle is read-only",
+			},
+			"rights": &schema.Schema{
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "list of rights assigned to this rights bundle",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"publish_to_all_tenants": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "When true, publishes the rights bundle to all tenants",
+			},
+			"tenants": &schema.Schema{
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "list of tenants to which this rights bundle is published",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+func resourceRightsBundleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	rightsBundleName := d.Get("name").(string)
+	publishToAllTenants := d.Get("publish_to_all_tenants").(bool)
+
+	inputRights, err := getRights(vcdClient, nil, "rights bundle create", d)
+	if err != nil {
+		return diag.Errorf("%s", err)
+	}
+	rightsBundle, err := vcdClient.Client.CreateRightsBundle(&types.RightsBundle{
+		Name:        rightsBundleName,
+		Description: d.Get("description").(string),
+		BundleKey:   types.VcloudUndefinedKey,
+		PublishAll:  takeBoolPointer(publishToAllTenants),
+	})
+	if err != nil {
+		return diag.Errorf("[rights bundle create] error creating role %s: %s", rightsBundleName, err)
+	}
+	if len(inputRights) > 0 {
+		err = rightsBundle.AddRights(inputRights)
+		if err != nil {
+			return diag.Errorf("[rights bundle create] error adding rights to rights bundle %s: %s", rightsBundleName, err)
+		}
+	}
+
+	inputTenants, err := getTenants(vcdClient, "rights bundle create", d)
+	if err != nil {
+		return diag.Errorf("%s", err)
+	}
+	if publishToAllTenants {
+		err = rightsBundle.PublishAllTenants()
+		if err != nil {
+			return diag.Errorf("[rights bundle create] error publishing to all tenants - rights bundle %s: %s", rightsBundleName, err)
+		}
+	}
+	if len(inputTenants) > 0 {
+		err = rightsBundle.PublishTenants(inputTenants)
+		if err != nil {
+			return diag.Errorf("[rights bundle create] error publishing to tenants - rights bundle %s: %s", rightsBundleName, err)
+		}
+	}
+	return genericRightsBundleRead(ctx, d, meta, "resource", "create")
+}
+
+func resourceRightsBundleRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return genericRightsBundleRead(ctx, d, meta, "resource", "read")
+}
+
+func genericRightsBundleRead(ctx context.Context, d *schema.ResourceData, meta interface{}, origin, operation string) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	rightsBundleName := d.Get("name").(string)
+
+	rightsBundle, err := vcdClient.Client.GetRightsBundleByName(rightsBundleName)
+	if err != nil {
+		if origin == "resource" {
```

Comment:
Similar safeguard as in globalRole
```suggestion
		if govcd.ContainsNotFound(err) && origin == "resource" {
```